CougSat1-PMIC/src/events/EventPeriodic.cpp

```diff
@@ -1,12 +1,247 @@
 #include "Events.h"
-
 #include <CISConsole.h>
+#include <PMICObjects.h>
+
+
+
+/**
+ * @brief Shed current from the nodes based on their rank
+ * Node with the largest (current * rank) is first to be shed
+ *
+ * @param nodes to shed from
+ * @param nodeCount of the nodes
+ * @param currentToShed in amps
+ * @return mbed_error_status_t
+ */
+mbed_error_status_t shedLoad(
+  CurrentNode * nodes[], uint8_t nodeCount, double currentToShed) 
+{
+      double              buf   = 0.0;
+      int                 count = 0;
+      mbed_error_status_t error = MBED_SUCCESS;
+
+      while (currentToShed > 0.0) 
+      {
+        // Could get stuck in this loop if all nodes are off but there is more
+        // current to shed TODO fix this
+        double        maxAggregate = 0.0;
+        CurrentNode * maxNode      = nullptr;
+
+        // Find the node with the largest aggregate rank that is enabled
+        for (int i = 0; i < nodeCount; ++i)
+        {
+            buf = nodes[i]->getAggregateRank();
+            if (buf > maxAggregate && nodes[i]->getSwitch())
```

Comment:
if (getSwitch()){
++count;
if(buf>maxAggregate){
max = buf
}
}
after the for loop, count is number of enabled nodes. if 0 return.